internal/visualization/console/plugin.go

```diff
@@ -9,9 +9,9 @@ import (
 )
 
 // ReconcilePlugin reconciles the console plugin to expose log querying of storage
-func ReconcilePlugin(k8sClient client.Client, logStore *logging.LogStoreSpec, owner client.Object, clusterVersion string) error {
+func ReconcilePlugin(k8sClient client.Client, logStore *logging.LogStoreSpec, owner client.Object, clusterVersion string, consoleSpec *logging.OCPConsoleSpec) error {
 	lokiService := lokistack.LokiStackGatewayService(logStore)
-	r := NewReconciler(k8sClient, NewConfig(owner, lokiService, FeaturesForOCP(clusterVersion)))
+	r := NewReconciler(k8sClient, NewConfig(owner, lokiService, FeaturesForOCP(clusterVersion)), consoleSpec)
 	if logStore != nil && logStore.Type == logging.LogStoreTypeLokiStack {
```

Comment:
Should the visualization still also depend on the value of `logStore` now that there is an explicit `spec.visualization.type` value?

Reply:
yes it has to be since we only deploy the plugin when "loki" is being used for the store.  Similar to kibana, deploying the visualization does not make sense if you do not have a back-end to query.

Reply:
Similar to the other thread: my idea was more going into the other direction: making the ocp-console an explicit opt-in for LokiStack instead of automatically activating it, just like Kibana was before with ES.

Reply:
@alanconway this is your call about explicit opt-in.  My concern would be that now it is not 'opt-in' which is a change in behavior of what already exists for 5.6

Reply:
I'd prefer to leave it as an automatic default, so there are fewer ways to end  up at the "why can't I see any logs" support call.